bin/citrea/tests/e2e/syncing.rs

```diff
@@ -396,9 +396,10 @@ async fn test_prover_sync_with_commitments() -> Result<(), anyhow::Error> {
 
     let da_data = blob.data.accumulator();
 
-    let proof: DaData = borsh::BorshDeserialize::try_from_slice(da_data).unwrap();
+    let data: DaDataLightClient = borsh::BorshDeserialize::try_from_slice(da_data).unwrap();
 
-    assert!(matches!(proof, DaData::ZKProof(_)));
+    // Test we got zkproof indeed
+    let DaDataLightClient::ZKProof(_proof) = data;
```

Comment:
What would happen if data is not `ZKProof` variant?

Reply:
Ah it works because that's the only variant in the enum:

https://github.com/chainwayxyz/citrea/pull/1037/files/3769a0f18a9e33bc2c67d02be2da152d01b285e3#diff-3bed783072482428302e286e30066c4b694af17ff1bc99d0ccd4e5f43fb2553eR48

If it wasn't, this won't compile:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bf51c18c2601e2b96e5209668e162d99